photon-api/src/main/scala/com/linkedin/photon/ml/estimators/GameEstimator.scala

```diff
@@ -0,0 +1,252 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.estimators
+
+import org.apache.spark.SparkContext
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.rdd.RDD
+import org.slf4j.Logger
+
+import com.linkedin.photon.ml.TaskType
+import com.linkedin.photon.ml.TaskType.TaskType
+import com.linkedin.photon.ml.Types.CoordinateId
+import com.linkedin.photon.ml.algorithm._
+import com.linkedin.photon.ml.constants.StorageLevel
+import com.linkedin.photon.ml.data._
+import com.linkedin.photon.ml.evaluation.Evaluator.EvaluationResults
+import com.linkedin.photon.ml.evaluation._
+import com.linkedin.photon.ml.function.glm._
+import com.linkedin.photon.ml.function.svm.{DistributedSmoothedHingeLossFunction, SingleNodeSmoothedHingeLossFunction}
+import com.linkedin.photon.ml.function.{DistributedObjectiveFunction, SingleNodeObjectiveFunction}
+import com.linkedin.photon.ml.model.GameModel
+import com.linkedin.photon.ml.normalization.{NoNormalization, NormalizationContext}
+import com.linkedin.photon.ml.optimization.DistributedOptimizationProblem
+import com.linkedin.photon.ml.optimization.game._
+import com.linkedin.photon.ml.sampler.{BinaryClassificationDownSampler, DefaultDownSampler, DownSampler}
+import com.linkedin.photon.ml.supervised.classification.{LogisticRegressionModel, SmoothedHingeLossLinearSVMModel}
+import com.linkedin.photon.ml.supervised.regression.{LinearRegressionModel, PoissonRegressionModel}
+import com.linkedin.photon.ml.util.Implicits._
+import com.linkedin.photon.ml.util._
+
+/**
+ * Estimator implementation for GAME models.
+ *
+ * @param sc The spark context for the application
+ * @param taskType The training task for the estimator
+ * @param numIterations The number of coordinate descent iterations to perform during estimation
+ * @param computeVariance Whether to compute the coefficient variances for estimated GAME models
+ * @param logger The logger instance for the application
+ */
+class GameEstimator(
+    val sc: SparkContext,
+    val taskType: TaskType,
+    val numIterations: Int,
+    val computeVariance: Boolean,
+    implicit val logger: Logger) {
+
+  import GameEstimator._
+
+  // 2 types that makes the code more readable
+  type LossFunction = (PointwiseLossFunction) => SingleNodeGLMLossFunction
+  type DistributedLossFunction = (PointwiseLossFunction) => DistributedGLMLossFunction
+
+  private val defaultNormalizationContext: Broadcast[NormalizationContext] = sc.broadcast(NoNormalization())
+
+  /**
+   * Fit a GAME model with the given configuration to the given training data.
+   *
```

Comment:
Add doc blurb here to explain strategy with iteration over coordinates and what not.